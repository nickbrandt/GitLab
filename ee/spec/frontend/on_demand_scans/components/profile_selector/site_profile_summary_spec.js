import { shallowMount } from '@vue/test-utils';
import App from 'ee/on_demand_scans/components/profile_selector/site_profile_summary.vue';
import { siteProfiles } from 'ee_jest/on_demand_scans/mocks/mock_data';

describe('DastSiteProfileSummary', () => {
  let wrapper;

  const createWrapper = (profile = {}) => {
    wrapper = shallowMount(App, {
      propsData: {
        profile,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('renders properly', () => {
    it.each(siteProfiles)('profile %#', (profile) => {
      createWrapper(profile);

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
