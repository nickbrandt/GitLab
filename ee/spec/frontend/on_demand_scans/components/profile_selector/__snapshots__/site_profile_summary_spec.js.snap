// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DastSiteProfileSummary renders properly profile 0 1`] = `
<div>
  <div
    class="row"
  >
    <profile-selector-summary-cell-stub
      class=""
      label="Target URL"
      value="https://foo.com"
    />
     
    <profile-selector-summary-cell-stub
      label="Site type"
      value="Website"
    />
  </div>
   
  <div
    class="row"
  >
    <profile-selector-summary-cell-stub
      label="Authentication URL"
      value="https://foo.com/login"
    />
  </div>
   
  <div
    class="row"
  >
    <profile-selector-summary-cell-stub
      label="Username"
      value="admin"
    />
     
    <profile-selector-summary-cell-stub
      label="Password"
      value="••••••••"
    />
  </div>
   
  <div
    class="row"
  >
    <profile-selector-summary-cell-stub
      label="Username form field"
      value="username"
    />
     
    <profile-selector-summary-cell-stub
      label="Password form field"
      value="password"
    />
  </div>
   
  <div
    class="row"
  >
    <profile-selector-summary-cell-stub
      label="Excluded URLs"
      value="https://foo.com/logout,https://foo.com/send_mail"
    />
     
    <profile-selector-summary-cell-stub
      label="Request headers"
      value="[Redacted]"
    />
  </div>
   
  <div
    class="row"
  >
    <profile-selector-summary-cell-stub
      label="Validation status"
      value="Not Validated"
    />
  </div>
</div>
`;

exports[`DastSiteProfileSummary renders properly profile 1 1`] = `
<div>
  <div
    class="row"
  >
    <profile-selector-summary-cell-stub
      class=""
      label="Target URL"
      value="https://bar.com"
    />
     
    <profile-selector-summary-cell-stub
      label="Site type"
      value="Rest API"
    />
  </div>
   
  <!---->
   
  <div
    class="row"
  >
    <profile-selector-summary-cell-stub
      label="Excluded URLs"
      value="https://bar.com/logout"
    />
     
    <profile-selector-summary-cell-stub
      label="Request headers"
      value="[Redacted]"
    />
  </div>
   
  <div
    class="row"
  >
    <profile-selector-summary-cell-stub
      label="Validation status"
      value="Validated"
    />
  </div>
</div>
`;
